src/test/java/com/commercetools/sync/sdk2/producttypes/utils/AttributeDefinitionUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,840 @@
+                .newValue(AttributeConstraintEnumDraft.NONE)
+                .build());
+  }
+
+  @Test
+  void buildChangeAttributeConstraintAction_WithSameValues_ShouldReturnEmptyOptional()
+      throws UnsupportedOperationException {
+    // Preparation
+    final AttributeDefinitionDraft draft =
+        createMockAttributeDefinitionDraftBuilder()
+            .attributeConstraint(AttributeConstraintEnum.COMBINATION_UNIQUE)
+            .build();
+
+    final AttributeDefinition attributeDefinition =
+        createMockAttributeDefinitionBuilder()
+            .attributeConstraint(AttributeConstraintEnum.COMBINATION_UNIQUE)
+            .build();
+
+    // test
+    final Optional<ProductTypeUpdateAction> result =
+        buildChangeAttributeConstraintUpdateAction(attributeDefinition, draft);
+
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  void
+      buildChangeAttributeConstraintAction_WithSourceNullValuesAndNonDefaultTarget_ShouldBuildAction()
+          throws UnsupportedOperationException {
+    // Preparation
+    final AttributeDefinitionDraft draft =
+        createMockAttributeDefinitionDraftBuilder().attributeConstraint(null).build();
+
+    final AttributeDefinition attributeDefinition =
+        createMockAttributeDefinitionBuilder()
+            .attributeConstraint(AttributeConstraintEnum.SAME_FOR_ALL)
+            .build();
+
+    // test
+    final Optional<ProductTypeUpdateAction> result =
+        buildChangeAttributeConstraintUpdateAction(attributeDefinition, draft);
+
+    assertThat(result)
+        .contains(
+            ProductTypeChangeAttributeConstraintActionBuilder.of()
+                .attributeName(draft.getName())
+                .newValue(AttributeConstraintEnumDraft.NONE)
+                .build());
+  }
+
+  @Test
+  void
+      buildChangeAttributeConstraintAction_WithSourceSameForAllAndTargetCombinationUnique_ShouldThrowException() {
+    // Preparation
+    final AttributeDefinition attributeDefinition =
+        createMockAttributeDefinitionBuilder()
+            .attributeConstraint(AttributeConstraintEnum.SAME_FOR_ALL)
+            .build();
+
+    final AttributeDefinitionDraft draft =
+        createMockAttributeDefinitionDraftBuilder()
+            .attributeConstraint(AttributeConstraintEnum.COMBINATION_UNIQUE)
+            .build();
+    // test
+    try {
+      buildChangeAttributeConstraintUpdateAction(attributeDefinition, draft);
+      fail("Expected an exception to be thrown");
+    } catch (UnsupportedOperationException e) {
+      assertThat(e)
+          .hasMessage(
+              "Invalid AttributeConstraint update to COMBINATION_UNIQUE. Only following updates are allowed: SameForAll to None and Unique to None.");
+    }
+  }
+
+  @Test
+  void buildActions_WithNewDifferentValues_ShouldReturnActions()
+      throws UnsupportedOperationException {
+    final List<ProductTypeUpdateAction> result = buildActions(old, newDifferent);
```

Comment:
As i see, here and below test is the only usage of `old, newDifferent and newSame`. So, the BeforeAll isn't necessary and instantiating it here would make it easier to read.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/989/commits/664dfb5e47b8220a5bc39a2057d542c7a5de5e8f